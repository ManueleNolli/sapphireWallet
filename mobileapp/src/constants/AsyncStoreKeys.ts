const theme = 'theme'
const firstAccess = 'firstAccess'

const EOAAddress = 'EOAAddress'

const walletContractAddress = 'walletContractAddress'

export { theme, firstAccess, EOAAddress, walletContractAddress }
