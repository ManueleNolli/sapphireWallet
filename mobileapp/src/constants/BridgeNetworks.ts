export enum BRIDGE_NETWORKS {
  MUMBAI = 'mumbai',
}
