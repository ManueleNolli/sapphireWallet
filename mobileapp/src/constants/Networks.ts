export enum NETWORKS {
  LOCALHOST = 'Localhost',
  SEPOLIA = 'Sepolia',
}
