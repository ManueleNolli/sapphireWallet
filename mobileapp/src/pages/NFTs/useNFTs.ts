import { useContext, useEffect, useState } from 'react'
import { BlockchainContext } from '../../context/BlockchainContext'
import { WalletContext } from '../../context/WalletContext'
import useLoading from '../../hooks/useLoading'
import { OwnedNFT, ownedNFTs } from '../../services/blockchain'

export default function useNFTs() {
  const { currentNetwork, ethersProvider } = useContext(BlockchainContext)
  const { getWalletContractAddress } = useContext(WalletContext)
  const { isLoading, setIsLoading } = useLoading(false)
  const { isLoading: isRefreshing, setIsLoading: setRefreshing } = useLoading(false)

  const [nfts, setNFTs] = useState<OwnedNFT[]>([])

  const getNFTs = async () => {
    return await ownedNFTs(getWalletContractAddress(), currentNetwork)
  }

  const refreshNFTs = async () => {
    setRefreshing(true)
    let nfts = await getNFTs()
    setNFTs(nfts)
    setRefreshing(false)
  }

  useEffect(() => {
    setIsLoading(true)
    getNFTs()
      .then((nfts) => {
        setNFTs(nfts)
      })
      .catch((err) => setNFTs([]))
      .finally(() => setIsLoading(false))
  }, [ethersProvider])

  return {
    isLoading,
    nfts,
    isRefreshing,
    refreshNFTs,
  }
}
