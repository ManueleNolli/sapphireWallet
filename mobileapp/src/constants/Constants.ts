import * as asyncStoreKeys from './AsyncStoreKeys'
import * as secureStoreKeys from './SecureStoreKeys'

export default {
  asyncStoreKeys,
  secureStoreKeys,
}
