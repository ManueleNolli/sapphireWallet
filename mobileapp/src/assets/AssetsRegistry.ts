import { NETWORKS } from '../constants/Networks'

const assetsPath = '../../assets/'

const attention = require(assetsPath + 'Attention.png')
const logo = require(assetsPath + 'Sapphire.png')
const logoAnimated = require(assetsPath + 'SapphireAnimated.gif')
const logoWithFullText = require(assetsPath + 'LogoFullText.png')

const homeBackground1 = require(assetsPath + 'HomeBackground.jpg')
const homeBackground2 = require(assetsPath + 'HomeBackground2.jpg')
const homeBackground3 = require(assetsPath + 'HomeBackground3.jpg')

const homeBackground = () => {
  const backgrounds = [homeBackground1, homeBackground2, homeBackground3]
  const randomIndex = Math.floor(Math.random() * backgrounds.length)
  return backgrounds[randomIndex]
}

const hardhatNetworkLogo = require(assetsPath + 'NetworkLogoHardhat.png')
const sepoliaNetworkLogo = require(assetsPath + 'NetworkLogoSepolia.png')
const ethereumNetworkLogo = require(assetsPath + 'NetworkLogoEthereum.png')

const networkLogo = (network: NETWORKS) => {
  switch (network) {
    case NETWORKS.LOCALHOST:
      return hardhatNetworkLogo
    case NETWORKS.SEPOLIA:
      return sepoliaNetworkLogo
    default:
      return ethereumNetworkLogo
  }
}

const qrCode = require(assetsPath + 'QrCode.png')
const sendETH = require(assetsPath + 'SendETH.png')
const sendNFTs = require(assetsPath + 'SendNFTs.png')

const NFTPlaceholder = require(assetsPath + 'NFTPlaceholder.png')

export {
  attention,
  logo,
  logoAnimated,
  logoWithFullText,
  homeBackground,
  networkLogo,
  qrCode,
  sendETH,
  sendNFTs,
  NFTPlaceholder,
}
