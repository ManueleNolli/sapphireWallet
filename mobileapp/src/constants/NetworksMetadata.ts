import { NETWORKS } from './Networks'
import { BRIDGE_NETWORKS } from './BridgeNetworks'

const ALL_NETWORKS = { ...NETWORKS, ...BRIDGE_NETWORKS }

type ChainIdsToNetwork = {
  [key: string]: (typeof ALL_NETWORKS)[keyof typeof ALL_NETWORKS]
}

export const CHAIN_IDS_TO_NETWORK: ChainIdsToNetwork = {
  '1337': ALL_NETWORKS.LOCALHOST,
  '11155111': ALL_NETWORKS.SEPOLIA,
  '80001': ALL_NETWORKS.MUMBAI,
}

type NetworkToChainIds = {
  [key: string]: string
}

export const NETWORK_TO_CHAIN_IDS: NetworkToChainIds = {
  [ALL_NETWORKS.LOCALHOST]: '1337',
  [ALL_NETWORKS.SEPOLIA]: '11155111',
  [ALL_NETWORKS.MUMBAI]: '80001',
}

type ChainCryptos = {
  [key: string]: string
}

export const CHAIN_CRYPTOS: ChainCryptos = {
  '1337': 'ETH',
  '11155111': 'ETH',
  '80001': 'MATIC',
}
