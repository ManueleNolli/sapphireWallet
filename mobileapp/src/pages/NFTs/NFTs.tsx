import React, { useContext, useEffect, useState } from 'react'
import SafeAreaView from '../../utils/SafeAreaView'
import { NFTCard } from '../../components/NFTCard/NFTCard'
import { NETWORKS } from '../../constants/Networks'
import { ScrollView, View } from 'react-native'
import { appStyles, vh } from '../../Styles'
import { Layout, Spinner, Text } from '@ui-kitten/components'
import useNFTs from './useNFTs'

export default function NFTs() {
  const { isLoading, nfts } = useNFTs()

  return (
    <Layout style={{ flex: 1 }}>
      <ScrollView>
        <SafeAreaView
          style={{
            paddingTop: 5 * vh,
            marginBottom: 0,
          }}
        >
          {isLoading && (
            <View style={appStyles.center}>
              <Spinner
                size="giant"
                status="basic"
                style={{
                  marginBottom: 2 * vh,
                }}
              />
            </View>
          )}
          {nfts.length === 0 && !isLoading && (
            <View style={appStyles.center}>
              <Text category={'label'}>No NFTs found</Text>
            </View>
          )}
          {nfts.map(
            (
              nft: {
                name: string
                tokenId: string
                collectionName: string
                collectionDescription: string
                image: string
                network: NETWORKS
              },
              index: number
            ) => (
              <NFTCard
                key={index}
                name={nft.name}
                tokenId={nft.tokenId}
                collectionName={nft.collectionName}
                collectionDescription={nft.collectionDescription}
                image={nft.image}
                network={nft.network}
                style={{ marginBottom: 2 * vh }}
              />
            )
          )}
        </SafeAreaView>
      </ScrollView>
    </Layout>
  )
}
