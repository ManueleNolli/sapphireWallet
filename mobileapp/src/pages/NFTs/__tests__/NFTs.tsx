import useNFTs from '../useNFTs'
import { NETWORKS } from '../../../constants/Networks'
import { NFTPlaceholder } from '../../../assets/AssetsRegistry'
import renderWithTheme from '../../../TestHelper'
import NFTs from '../NFTs'
import React from 'react'

jest.mock('../useNFTs', () => jest.fn())
jest.mock('@ui-kitten/components', () => {
  const { View } = require('react-native')
  return {
    ...jest.requireActual('@ui-kitten/components'),
    Spinner: () => <View data-testid="spinner" />,
  }
})
describe('NFTs', () => {
  beforeEach(() => {
    jest.clearAllMocks()
  })
  it('render correctly', () => {
    ;(useNFTs as jest.Mock).mockReturnValue({
      isLoading: false,
      nfts: [
        {
          name: 'name0',
          description: 'description0',
          image: NFTPlaceholder,
          tokenId: 0,
          network: NETWORKS.LOCALHOST,
          collectionAddress: 'collectionAddress0',
          collectionName: 'collectionName0',
          collectionDescription: 'collectionDescription0',
        },
        {
          name: 'name1',
          description: 'description1',
          image: NFTPlaceholder,
          tokenId: 1,
          network: NETWORKS.LOCALHOST,
          collectionAddress: 'collectionAddress1',
          collectionName: 'collectionName1',
          collectionDescription: 'collectionDescription1',
        },
      ],
    })

    const tree = renderWithTheme(<NFTs />)

    expect(tree).toMatchSnapshot()
  })

  it('render correctly without nfts', () => {
    ;(useNFTs as jest.Mock).mockReturnValue({
      isLoading: false,
      nfts: [],
    })

    const tree = renderWithTheme(<NFTs />)

    expect(tree.getAllByText('No NFTs found')).toBeTruthy()
    expect(tree).toMatchSnapshot()
  })

  it('render correctly loading', () => {
    ;(useNFTs as jest.Mock).mockReturnValue({
      isLoading: true,
      nfts: [],
    })

    const tree = renderWithTheme(<NFTs />)

    expect(tree).toMatchSnapshot()
  })
})
