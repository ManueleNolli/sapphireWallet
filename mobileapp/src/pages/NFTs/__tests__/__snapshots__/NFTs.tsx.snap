// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NFTs render correctly 1`] = `
<View
  appearance="default"
  level="1"
  style={
    [
      {
        "backgroundColor": "#FFFFFF",
      },
      {
        "flex": 1,
      },
    ]
  }
>
  <RCTScrollView>
    <View>
      <View
        appearance="default"
        level="1"
        style={
          [
            {
              "backgroundColor": "#FFFFFF",
            },
            {
              "flex": 1,
            },
          ]
        }
      >
        <RNCSafeAreaView
          edges={
            {
              "bottom": "additive",
              "left": "additive",
              "right": "additive",
              "top": "additive",
            }
          }
          style={
            [
              {
                "flex": 1,
                "marginHorizontal": 37.5,
                "marginVertical": 66.7,
              },
              {
                "marginBottom": 0,
                "paddingTop": 66.7,
              },
            ]
          }
        >
          <View
            style={
              [
                {
                  "borderColor": "#598BFF",
                  "borderRadius": 10,
                  "borderWidth": 1,
                  "height": 667,
                  "overflow": "hidden",
                  "width": "100%",
                },
                {
                  "marginBottom": 26.68,
                },
              ]
            }
          >
            <View
              style={
                {
                  "elevation": 15,
                  "flex": 1,
                  "flexDirection": "column-reverse",
                  "overflow": "hidden",
                  "shadowColor": "black",
                  "shadowOffset": {
                    "height": 0,
                    "width": 0,
                  },
                  "shadowOpacity": 0.5,
                  "shadowRadius": 10,
                }
              }
            >
              <ViewManagerAdapter_ExpoImage
                bottom={0}
                contentFit="cover"
                contentPosition={
                  {
                    "left": "50%",
                    "top": "50%",
                  }
                }
                left={0}
                onError={[Function]}
                onLoad={[Function]}
                onLoadStart={[Function]}
                onProgress={[Function]}
                placeholder={[]}
                position="absolute"
                right={0}
                source={[]}
                style={
                  {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
                top={0}
                transition={null}
              />
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "borderRadius": 5,
                    "marginHorizontal": 15,
                    "marginVertical": 13.34,
                    "opacity": 1,
                    "overflow": "hidden",
                  }
                }
                testID="CollapsibleButton"
              >
                <View
                  style={
                    [
                      {
                        "backgroundColor": "transparent",
                      },
                      {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "justifyContent": "space-between",
                        "maxHeight": 600.3,
                        "paddingHorizontal": 15,
                        "paddingVertical": 13.34,
                      },
                    ]
                  }
                >
                  <ViewManagerAdapter_ExpoBlurView
                    blurReductionFactor={4}
                    intensity={50}
                    style={
                      {
                        "bottom": 0,
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      }
                    }
                    tint="default"
                  />
                  <View
                    style={
                      {
                        "width": "90%",
                      }
                    }
                  >
                    <Text
                      appearance="default"
                      category="h4"
                      status="basic"
                      style={
                        [
                          {
                            "color": "#222B45",
                            "fontFamily": "System",
                            "fontSize": 26,
                            "fontWeight": "800",
                          },
                          undefined,
                        ]
                      }
                    >
                      name0
                    </Text>
                    <View
                      collapsable={false}
                      pointerEvents="none"
                      style={
                        {
                          "height": 0,
                          "overflow": "hidden",
                        }
                      }
                    >
                      <View
                        collapsable={false}
                        onLayout={[Function]}
                        style={{}}
                      >
                        <View>
                          <View
                            style={
                              {
                                "alignItems": "center",
                                "flexDirection": "row",
                                "marginVertical": 26.68,
                              }
                            }
                          >
                            <ViewManagerAdapter_ExpoImage
                              contentFit="contain"
                              contentPosition={
                                {
                                  "left": "50%",
                                  "top": "50%",
                                }
                              }
                              height={24}
                              marginRight={8}
                              onError={[Function]}
                              onLoad={[Function]}
                              onLoadStart={[Function]}
                              onProgress={[Function]}
                              placeholder={[]}
                              source={[]}
                              style={
                                {
                                  "height": 24,
                                  "marginRight": 8,
                                  "width": 24,
                                }
                              }
                              transition={null}
                              width={24}
                            />
                            <Text
                              appearance="default"
                              category="label"
                              status="basic"
                              style={
                                [
                                  {
                                    "color": "#222B45",
                                    "fontFamily": "System",
                                    "fontSize": 12,
                                    "fontWeight": "800",
                                  },
                                  undefined,
                                ]
                              }
                            >
                              localhost
                            </Text>
                          </View>
                          <Text
                            appearance="default"
                            category="h6"
                            status="basic"
                            style={
                              [
                                {
                                  "color": "#222B45",
                                  "fontFamily": "System",
                                  "fontSize": 18,
                                  "fontWeight": "800",
                                },
                                undefined,
                              ]
                            }
                          >
                            collectionName0
                          </Text>
                          <Text
                            appearance="default"
                            category="p1"
                            status="basic"
                            style={
                              [
                                {
                                  "color": "#222B45",
                                  "fontFamily": "System",
                                  "fontSize": 15,
                                  "fontWeight": "400",
                                },
                                undefined,
                              ]
                            }
                          >
                            collectionDescription0
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                  <View
                    style={
                      {
                        "height": "100%",
                      }
                    }
                  >
                    <View
                      collapsable={false}
                      style={
                        {
                          "transform": [
                            {
                              "scale": 1,
                            },
                          ],
                        }
                      }
                    >
                      <RNSVGSvgView
                        align="xMidYMid"
                        bbHeight="32"
                        bbWidth="32"
                        fill="#222B45"
                        focusable={false}
                        meetOrSlice={0}
                        minX={0}
                        minY={0}
                        style={
                          [
                            {
                              "backgroundColor": "transparent",
                              "borderWidth": 0,
                            },
                            {
                              "height": 32,
                              "tintColor": "#222B45",
                              "width": 32,
                            },
                            {
                              "flex": 0,
                              "height": 32,
                              "width": 32,
                            },
                          ]
                        }
                        vbHeight={24}
                        vbWidth={24}
                      >
                        <RNSVGGroup
                          fill={
                            {
                              "payload": 4280429381,
                              "type": 0,
                            }
                          }
                          propList={
                            [
                              "fill",
                            ]
                          }
                        >
                          <RNSVGGroup
                            fill={
                              {
                                "payload": 4278190080,
                                "type": 0,
                              }
                            }
                          >
                            <RNSVGGroup
                              fill={
                                {
                                  "payload": 4278190080,
                                  "type": 0,
                                }
                              }
                            >
                              <RNSVGRect
                                fill={
                                  {
                                    "payload": 4278190080,
                                    "type": 0,
                                  }
                                }
                                height="24"
                                matrix={
                                  [
                                    -1,
                                    1.2246467991473532e-16,
                                    -1.2246467991473532e-16,
                                    -1,
                                    24,
                                    24,
                                  ]
                                }
                                opacity={0}
                                width="24"
                                x="0"
                                y="0"
                              />
                              <RNSVGPath
                                d="M18 15a1 1 0 0 1-.64-.23L12 10.29l-5.37 4.32a1 1 0 0 1-1.41-.15 1 1 0 0 1 .15-1.41l6-4.83a1 1 0 0 1 1.27 0l6 5a1 1 0 0 1 .13 1.41A1 1 0 0 1 18 15z"
                                fill={
                                  {
                                    "payload": 4278190080,
                                    "type": 0,
                                  }
                                }
                              />
                            </RNSVGGroup>
                          </RNSVGGroup>
                        </RNSVGGroup>
                      </RNSVGSvgView>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              [
                {
                  "borderColor": "#598BFF",
                  "borderRadius": 10,
                  "borderWidth": 1,
                  "height": 667,
                  "overflow": "hidden",
                  "width": "100%",
                },
                {
                  "marginBottom": 26.68,
                },
              ]
            }
          >
            <View
              style={
                {
                  "elevation": 15,
                  "flex": 1,
                  "flexDirection": "column-reverse",
                  "overflow": "hidden",
                  "shadowColor": "black",
                  "shadowOffset": {
                    "height": 0,
                    "width": 0,
                  },
                  "shadowOpacity": 0.5,
                  "shadowRadius": 10,
                }
              }
            >
              <ViewManagerAdapter_ExpoImage
                bottom={0}
                contentFit="cover"
                contentPosition={
                  {
                    "left": "50%",
                    "top": "50%",
                  }
                }
                left={0}
                onError={[Function]}
                onLoad={[Function]}
                onLoadStart={[Function]}
                onProgress={[Function]}
                placeholder={[]}
                position="absolute"
                right={0}
                source={[]}
                style={
                  {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
                top={0}
                transition={null}
              />
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "borderRadius": 5,
                    "marginHorizontal": 15,
                    "marginVertical": 13.34,
                    "opacity": 1,
                    "overflow": "hidden",
                  }
                }
                testID="CollapsibleButton"
              >
                <View
                  style={
                    [
                      {
                        "backgroundColor": "transparent",
                      },
                      {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "justifyContent": "space-between",
                        "maxHeight": 600.3,
                        "paddingHorizontal": 15,
                        "paddingVertical": 13.34,
                      },
                    ]
                  }
                >
                  <ViewManagerAdapter_ExpoBlurView
                    blurReductionFactor={4}
                    intensity={50}
                    style={
                      {
                        "bottom": 0,
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      }
                    }
                    tint="default"
                  />
                  <View
                    style={
                      {
                        "width": "90%",
                      }
                    }
                  >
                    <Text
                      appearance="default"
                      category="h4"
                      status="basic"
                      style={
                        [
                          {
                            "color": "#222B45",
                            "fontFamily": "System",
                            "fontSize": 26,
                            "fontWeight": "800",
                          },
                          undefined,
                        ]
                      }
                    >
                      name1
                    </Text>
                    <View
                      collapsable={false}
                      pointerEvents="none"
                      style={
                        {
                          "height": 0,
                          "overflow": "hidden",
                        }
                      }
                    >
                      <View
                        collapsable={false}
                        onLayout={[Function]}
                        style={{}}
                      >
                        <View>
                          <View
                            style={
                              {
                                "alignItems": "center",
                                "flexDirection": "row",
                                "marginVertical": 26.68,
                              }
                            }
                          >
                            <ViewManagerAdapter_ExpoImage
                              contentFit="contain"
                              contentPosition={
                                {
                                  "left": "50%",
                                  "top": "50%",
                                }
                              }
                              height={24}
                              marginRight={8}
                              onError={[Function]}
                              onLoad={[Function]}
                              onLoadStart={[Function]}
                              onProgress={[Function]}
                              placeholder={[]}
                              source={[]}
                              style={
                                {
                                  "height": 24,
                                  "marginRight": 8,
                                  "width": 24,
                                }
                              }
                              transition={null}
                              width={24}
                            />
                            <Text
                              appearance="default"
                              category="label"
                              status="basic"
                              style={
                                [
                                  {
                                    "color": "#222B45",
                                    "fontFamily": "System",
                                    "fontSize": 12,
                                    "fontWeight": "800",
                                  },
                                  undefined,
                                ]
                              }
                            >
                              localhost
                            </Text>
                          </View>
                          <Text
                            appearance="default"
                            category="h6"
                            status="basic"
                            style={
                              [
                                {
                                  "color": "#222B45",
                                  "fontFamily": "System",
                                  "fontSize": 18,
                                  "fontWeight": "800",
                                },
                                undefined,
                              ]
                            }
                          >
                            collectionName1
                          </Text>
                          <Text
                            appearance="default"
                            category="p1"
                            status="basic"
                            style={
                              [
                                {
                                  "color": "#222B45",
                                  "fontFamily": "System",
                                  "fontSize": 15,
                                  "fontWeight": "400",
                                },
                                undefined,
                              ]
                            }
                          >
                            collectionDescription1
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                  <View
                    style={
                      {
                        "height": "100%",
                      }
                    }
                  >
                    <View
                      collapsable={false}
                      style={
                        {
                          "transform": [
                            {
                              "scale": 1,
                            },
                          ],
                        }
                      }
                    >
                      <RNSVGSvgView
                        align="xMidYMid"
                        bbHeight="32"
                        bbWidth="32"
                        fill="#222B45"
                        focusable={false}
                        meetOrSlice={0}
                        minX={0}
                        minY={0}
                        style={
                          [
                            {
                              "backgroundColor": "transparent",
                              "borderWidth": 0,
                            },
                            {
                              "height": 32,
                              "tintColor": "#222B45",
                              "width": 32,
                            },
                            {
                              "flex": 0,
                              "height": 32,
                              "width": 32,
                            },
                          ]
                        }
                        vbHeight={24}
                        vbWidth={24}
                      >
                        <RNSVGGroup
                          fill={
                            {
                              "payload": 4280429381,
                              "type": 0,
                            }
                          }
                          propList={
                            [
                              "fill",
                            ]
                          }
                        >
                          <RNSVGGroup
                            fill={
                              {
                                "payload": 4278190080,
                                "type": 0,
                              }
                            }
                          >
                            <RNSVGGroup
                              fill={
                                {
                                  "payload": 4278190080,
                                  "type": 0,
                                }
                              }
                            >
                              <RNSVGRect
                                fill={
                                  {
                                    "payload": 4278190080,
                                    "type": 0,
                                  }
                                }
                                height="24"
                                matrix={
                                  [
                                    -1,
                                    1.2246467991473532e-16,
                                    -1.2246467991473532e-16,
                                    -1,
                                    24,
                                    24,
                                  ]
                                }
                                opacity={0}
                                width="24"
                                x="0"
                                y="0"
                              />
                              <RNSVGPath
                                d="M18 15a1 1 0 0 1-.64-.23L12 10.29l-5.37 4.32a1 1 0 0 1-1.41-.15 1 1 0 0 1 .15-1.41l6-4.83a1 1 0 0 1 1.27 0l6 5a1 1 0 0 1 .13 1.41A1 1 0 0 1 18 15z"
                                fill={
                                  {
                                    "payload": 4278190080,
                                    "type": 0,
                                  }
                                }
                              />
                            </RNSVGGroup>
                          </RNSVGGroup>
                        </RNSVGGroup>
                      </RNSVGSvgView>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNCSafeAreaView>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`NFTs render correctly loading 1`] = `
<View
  appearance="default"
  level="1"
  style={
    [
      {
        "backgroundColor": "#FFFFFF",
      },
      {
        "flex": 1,
      },
    ]
  }
>
  <RCTScrollView>
    <View>
      <View
        appearance="default"
        level="1"
        style={
          [
            {
              "backgroundColor": "#FFFFFF",
            },
            {
              "flex": 1,
            },
          ]
        }
      >
        <RNCSafeAreaView
          edges={
            {
              "bottom": "additive",
              "left": "additive",
              "right": "additive",
              "top": "additive",
            }
          }
          style={
            [
              {
                "flex": 1,
                "marginHorizontal": 37.5,
                "marginVertical": 66.7,
              },
              {
                "marginBottom": 0,
                "paddingTop": 66.7,
              },
            ]
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <View
              data-testid="spinner"
            />
          </View>
        </RNCSafeAreaView>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`NFTs render correctly without nfts 1`] = `
<View
  appearance="default"
  level="1"
  style={
    [
      {
        "backgroundColor": "#FFFFFF",
      },
      {
        "flex": 1,
      },
    ]
  }
>
  <RCTScrollView>
    <View>
      <View
        appearance="default"
        level="1"
        style={
          [
            {
              "backgroundColor": "#FFFFFF",
            },
            {
              "flex": 1,
            },
          ]
        }
      >
        <RNCSafeAreaView
          edges={
            {
              "bottom": "additive",
              "left": "additive",
              "right": "additive",
              "top": "additive",
            }
          }
          style={
            [
              {
                "flex": 1,
                "marginHorizontal": 37.5,
                "marginVertical": 66.7,
              },
              {
                "marginBottom": 0,
                "paddingTop": 66.7,
              },
            ]
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              appearance="default"
              category="label"
              status="basic"
              style={
                [
                  {
                    "color": "#222B45",
                    "fontFamily": "System",
                    "fontSize": 12,
                    "fontWeight": "800",
                  },
                  undefined,
                ]
              }
            >
              No NFTs found
            </Text>
          </View>
        </RNCSafeAreaView>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
